Feat : DIG-92 채팅 기능 구현

YEJINGO opened the pull request:
- JwtUtils : StompHandler 에서 사용되는 토큰 검증 로직 추가
- Entity 수정
  - ChatRoom : 채팅방에 필요한 값 추가
  - ChatMessage : 채팅 메세지에 필요한 값 추가
- build.gradle : websocket,redis 추거
- ErrorCode : 400으로 되어있는 값들 404 로 변경

YEJINGO committed:
Feat : DIG-92 채팅 기능 구현

- JwtUtils : StompHandler 에서 사용되는 토큰 검증 로직 추가
- Entity 수정
  - ChatRoom : 채팅방에 필요한 값 추가
  - ChatMessage : 채팅 메세지에 필요한 값 추가
- build.gradle : websocket,redis 추거
- ErrorCode : 400으로 되어있는 값들 404 로 변경

YEJINGO committed:
Feat : DIG-92 채팅 기능 구현

- 좋아요/댓글,대댓글 OAuth2JwtUserDetails 추가

bstaran reviewed (approved):
너무너무 고생 많으셨습니다..! 👍 👍

bstaran commented:
제가 수정한 부분에서 충돌이 좀 있나봅니다.
미리 죄송합니다. 🥲

YEJINGO committed:
Feat : DIG-92 채팅 기능 구현

- StompHandler 수정 : CONNECT,SUBSCRIBE,DISCONNET 추가
- ChatMessageService
    - loadMessage 로직 수정 : redis에 저장된 메세지 값에도 인원수에 따라 readCount값을 수정, update된 redis값 추가 설정
    - updateReadCount 추가 : 채팅방에 들어온 인원수에 따라 readCount값을 set 한다. ex) 채팅방에 들어온 인원수가 2일경우 readCount는 0이되고, 그 외의 경우 readCount는 1이된다.
- JwtUtils : token값으로 email 가져오는 로직 추가
